controllers/licenseissuer/internal/manager/notification_mgr.go

```diff
@@ -113,6 +131,7 @@ func NotificationPackageToNotification(pack NotificationPackage) ntf.Notificatio
 	notification.Spec.From = string(pack.From)
 	notification.Spec.Message = string(pack.Message)
 	notification.Spec.Title = string(pack.Title)
+	notification.Spec.Importance = pack.Level
 	return notification
 }
 
```

Comment:
```go
func (nt *NotificationTask) Run() error {
	for _, data := range nt.NotificationCache {
		data.Namespace = nt.Target
		var tmp ntf.Notification
		err := nt.client.Get(nt.ctx, types.NamespacedName{Namespace: data.Namespace, Name: data.Name}, &tmp)
		if err == nil {
			continue
		} else if apierrors.IsNotFound(err) {
			err := nt.client.Create(nt.ctx, &data)
			if err == nil {
				continue
			}
			return err
		} else {
			return err
		}
	}
	return nil
}
```
fix this func, if create error and you return immediately, what about the notification left?

Reply:
"This function merely provides a supplementary notification feature after the main function has been accomplished. By design, it allows notification failures. Repeatedly delivering failed notifications would increase the maintenance and design costs of the code. The focus of the controller should be on ensuring the success of the main function."

Reply:
This is a callback function within a callback framework, where errors are handled within the framework itself.

Reply:
commonly we use `if err != nil`...